File: java sorting/SelectionSort.java
public class SelectionSort {
    public SelectionSort() {
    }

    public static void main(String[] args) {
        int[] arr = new int[]{10, 5, 8, 20, 2, 18};
        int n = arr.length;
        int[] temp = new int[n];

        int j;
        for(j = 0; j < n; ++j) {
            int min_ind = 0;

            for(int j = 1; j < n; ++j) {
                if (arr[j] < arr[min_ind]) {
                    min_ind = j;
                }
            }

            temp[j] = arr[min_ind];
            arr[min_ind] = 2147483647;
        }

        for(j = 0; j < n; ++j) {
            arr[j] = temp[j];
        }

        for(j = 0; j < n; ++j) {
            System.out.print(arr[j] + " ");
        }

    }
}

File: Selectionsort.cpp
#include <iostream>
using namespace std;
void disp(int arr[], int len)
{
    for (int i = 0; i < len; i++)
        cout << arr[i] << " ";
}
void sort(int arr[], int len) // function for selection sort
{

    int temp,count=0;
    for (int i = 0; i < len - 1; i++)
    {
        for (int j = i + 1; j < len; j++)
        {
            if (arr[j] < arr[i])
            {
                temp = arr[j];
                arr[j] = arr[i];
                arr[i] = temp;
                
    
               
            }
        
            
        }

    }
}
int main()
{
    cout<<"Initally the  elements are: ";
    int arr[] = {3,7,1,5,2};
    int len = 5;
 
    disp(arr, len);
    cout << endl
         << "Array after sorting: ";
    sort(arr, len);
  
    disp(arr, len);

}